win32/devdoc/job_object_helper.md

```diff
@@ -1,44 +1,123 @@
-# job_object_help requirements
+# job_object_helper requirements
 
-`job_object_help` is a module that wraps the Windows JobObject API, primarily for the purpose of limiting the amount of memory and CPU the current process can consume.
+`job_object_helper` is a module that wraps the Windows JobObject API, primarily for the purpose of limiting the amount of memory and CPU the current process can consume.
 
 ## Exposed API
 ```c
-    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_physical_memory, uint32_t, percent_cpu);
+MOCKABLE_FUNCTION(, THANDLE(JOB_OBJECT_HELPER), job_object_helper_create);
+MOCKABLE_FUNCTION(, int, job_object_helper_open, THANDLE(JOB_OBJECT_HELPER), job_object_helper);
+MOCKABLE_FUNCTION(, void, job_object_helper_close, THANDLE(JOB_OBJECT_HELPER), job_object_helper);
+MOCKABLE_FUNCTION(, int, job_object_helper_limit_memory, THANDLE(JOB_OBJECT_HELPER), job_object_helper, uint32_t, percent_physical_memory);
+MOCKABLE_FUNCTION(, int, job_object_helper_limit_cpu, THANDLE(JOB_OBJECT_HELPER), job_object_helper, uint32_t, percent_cpu);
 ```
 
-## job_object_helper_limit_resources
+## job_object_helper_create
 ```c
-    MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_physical_memory, uint32_t, percent_cpu);
+MOCKABLE_FUNCTION(, THANDLE(JOB_OBJECT_HELPER), job_object_helper_create);
 ```
+``job_object_helper_create`` is used to create a `JOB_OBJECT_HELPER` object.
 
-`job_object_helper_limit_resources` is an API that uses the job object API to limit the amount of physical memory and CPU that the current process can consume.
+**SRS_JOB_OBJECT_HELPER_18_016: [** `job_object_helper_create` shall allocate a `JOB_OBJECT_HELPER` object. **]**
 
-**SRS_JOB_OBJECT_HELPER_18_001: [** If `percent_physical_memory` is `0`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
+**SRS_JOB_OBJECT_HELPER_18_017: [** `job_object_helper_create` shall call `sm_create`. **]**

```

Comment:
>sm_create [](http://example.com/codeflow?start=75&length=9)

why is SM needed here? This module is 
1) THANDLEd
2) doesn't have callbacks.

So I suspect that sm is not needed.